services/docker/docker-size.service.js

```diff
@@ -20,14 +27,40 @@ const pagedSchema = Joi.object({
     Joi.object({
       name: Joi.string().required(),
       full_size: nonNegativeInteger.required(),
+      images: Joi.array().items(
+        Joi.object({
+          size: nonNegativeInteger.required(),
+          architecture: Joi.string().required(),
+        })
+      ),
     })
   ),
 }).required()
 
 const queryParamSchema = Joi.object({
   sort: Joi.string().valid('date', 'semver').default('date'),
+  arch: Joi.alternatives([
+    Joi.string().valid(...validArchitectures),
+    Joi.number().valid(...validArchitectures),
+  ]),
```

Comment:
I've put a comment in elsewhere saying I don't think we need to allow `architecture` to be a number at all. Assuming we can ditch it, this can just be `arch: Joi.string().valid(...validArchitectures)`. If not, lets come back to it.

Reply:
This is connected to my comment https://github.com/badges/shields/pull/8290#discussion_r951697982 
If needed, I will update this chunk of code :)

Reply:
Updated as per https://github.com/badges/shields/pull/8290#discussion_r954208573